src/openrct2/ride/Ride.cpp

```diff
@@ -3565,7 +3659,10 @@ void Ride::MoveTrainsToBlockBrakes(TrackElement* firstBlock)
             }
         } while (!(train->UpdateTrackMotion(nullptr) & VEHICLE_UPDATE_MOTION_TRACK_FLAG_VEHICLE_AT_BLOCK_BRAKE));
 
-        firstBlock->SetBlockBrakeClosed(true);
+        firstBlock->SetBrakeClosed(true);
+        blockBrakeSetLinkedBrakesClosed(
```

Comment:
Could add a check here to determine if it's a TrackElemType::BlockBrake

Reply:
It is always a block brake because VEHICLE_UPDATE_MOTION_TRACK_FLAG_VEHICLE_AT_BLOCK_BRAKE is only tripped when it's a block brake